dataclients/kubernetes/path_test.go

```diff
@@ -0,0 +1,392 @@
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/zalando/skipper/eskip"
+)
+
+func findPathPredicate(r *eskip.Route, name string) (*eskip.Predicate, error) {
+	var preds []*eskip.Predicate
+	for _, p := range r.Predicates {
+		if p.Name == name {
+			preds = append(preds, p)
+		}
+	}
+
+	if len(preds) > 1 {
+		return nil, fmt.Errorf("multiple predicates of the same name: %d %s", len(preds), name)
+	}
+
+	if len(preds) == 1 {
+		return preds[0], nil
+	}
+
+	return nil, nil
+}
+
+func findRouteWithRx(r []*eskip.Route) *eskip.Route {
+	for _, ri := range r {
+		if len(ri.PathRegexps) == 1 {
+			return ri
+		}
+	}
+
+	return nil
+}
+
+func findRouteWithPathPrefix(r []*eskip.Route) (*eskip.Route, error) {
+	for _, ri := range r {
+		p, err := findPathPredicate(ri, "PathSubtree")
+		if err != nil {
+			return nil, err
+		}
+
+		if p != nil {
+			return ri, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func findRouteWithExactPath(r []*eskip.Route) (*eskip.Route, error) {
```

Comment:
this can be dropped and all tests depdening on that, correct?

Reply:
this is used in the test where we use a path prefix for the ingress path, but the annotation contains an explicit path, and we verify that the explicit path overrules the path prefix